### PR TITLE
feat: junit5 support for TestBench

### PR DESCRIPTION
**Pull request by MarcinVaadin**
### Added `vaadin-testbench-core-junit5`
- Removed parallel scheduler in favor of native JUnit5 parallel support
- `DesiredCapabilities` features extracted to new package
- Multiple capabilities are now run using `TestTemplate` with `InvocationContextProvider`
- `Rules` converted to `Extensions`

### Added `vaadin-integration-tests-junit5` that are using new approach
- Prallel runs configured

### What needs to be done still:
- Implement `RetryRule` feature in JUnit5 way
### Migration notes:
- Update imports to `org.junit.jupiter.api.`
- Annotate test method with `@TestBenchTest` (name to be discussed) instead of `@Test`
- Extend `AbstractJUnit5TB6Test` instead of `AbstractTB6Test`

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by MarcinVaadin** [comment]
Please don't care those dozens of pushes above - I was investigating the reason of CI tests flakiness...

**Review by mshabarov — CHANGES_REQUESTED** [review]
I haven't checked test and view classes, because I assume they are 99% copy-pasted from core module.

**Comment by mshabarov** [comment]
Do you have, by the way, any tests/examples how can failsafe plugin be set up for retry on failures?

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetCapabilities.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.vaadin.testbench;
+
+import org.openqa.selenium.Capabilities;
+
+public interface SetCapabilities {
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetCapabilities.java:5]: There is already org.openqa.selenium.HasCapabilities I need a name for interface with setter. Suggestion?

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetDriver.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.vaadin.testbench;
+
+import org.openqa.selenium.WebDriver;
+
+public interface SetDriver {
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetDriver.java:5]: Yes, with empty `default` not to cause backward issues.

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/capabilities/CapabilitiesInvocationContextProvider.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(
+            ExtensionContext context) {
+        return CapabilitiesUtil.getDesiredCapabilities(context).stream()
+                .map(dc -> new DesireCapabilitiesTestTemplateInvocationContext(
+                        context, dc));
+    }
+
+    private static class DesireCapabilitiesTestTemplateInvocationContext
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/capabilities/CapabilitiesInvocationContextProvider.java:44]: Yes, it is possible to write custom `InvocationContextProvider` that will calculate custom displayName but will apply `CapabilitiesExtension` as it is already.

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchUtil.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import com.vaadin.testbench.commands.TestBenchCommandExecutor;
+import com.vaadin.testbench.commands.TestBenchCommands;
+
+public class TestBenchUtil
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchUtil.java]: Done

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchTestBase.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import com.vaadin.testbench.commands.TestBenchCommandExecutor;
+import com.vaadin.testbench.commands.TestBenchCommands;
+
+public abstract class TestBenchTestBase implements HasDriver, SetDriver,
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchTestBase.java:30]: Javadoc copied from `TestBenchTestCase`

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetCapabilities.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetCapabilities.java]: Removed in favor of ParameterResolver

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetDriver.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/SetDriver.java]: Removed in favor of ParameterResolver

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchTest.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import com.vaadin.testbench.capabilities.CapabilitiesInvocationContextProvider;
+
+/**
+ * Shorthand annotation that combines {@code @TestTemplate} together with
```
Comment (mshabarov) [inline vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchTest.java:23]:
```suggestion
 * Shorthand annotation for marking test methods in test class.
 * <p>
 * Combines JUnit 5 {@code @TestTemplate} together with
```

File: vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchTestBase.java [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import com.vaadin.testbench.commands.TestBenchCommandExecutor;
+import com.vaadin.testbench.commands.TestBenchCommands;
+
+/**
+ * A superclass with some helpers to aid TestBench developers.
```
Comment (mshabarov) [inline vaadin-testbench-core-junit5/src/main/java/com/vaadin/testbench/TestBenchTestBase.java:32]:
```suggestion
 * A superclass with helper methods to aid TestBench developers create a JUnit 5 based tests.
```

**Review by mshabarov — APPROVED** [review]

**Comment by mshabarov** [comment]
A couple of minor suggestions for javadoc, otherwise LGTM

**Pull request merged**